src/bindings/python/flux/job/JobID.py

```diff
@@ -53,15 +53,19 @@ class JobID(int):
     """
 
     def __new__(cls, value):
-        if isinstance(value, int):
-            jobid = value
+        if isinstance(value, JobID):
+            self = super(cls, cls).__new__(cls, int(value))
+            self.orig_str = value.orig
```

Comment:
This might be simpler for the copy constructor implementation:
```suggestion
        if isinstance(value, JobID):
            return cls(value.orig_str)
```

Then the rest of the function stays the same.

Reply:
oh yeah, that is much simpler :-)